kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -182,7 +187,7 @@ public long getVersion() {
 
   /** Returns the crc info for the current snapshot if the checksum file is read */
   public Optional<CRCInfo> getCurrentCrcInfo() {
```

Comment:
So this is introducing a similar complexity to what we had in TransactionImpl a few weeks back

We now have a private member field that we should NOT use -- and instead we should use the getter method -- but both of these are available in the class!

So we either need to (1) wrap these behind a hidden scope/context, or (2) set it once and only once.

Reply:
Good catch, likewise I define a private class